### ISSUE TITLE
Strip prefix from argument value if BDD keyword starts with embedded argument

### ISSUE THREAD

**Issue by allanlei** [robotframework/robotframework#4577]
When using `Given/When/Then/etc` to match keywords that start with a variable `${...}`, then `Give/When/Then` is captured into the variable value.

According to [Ignoring Given/When/Then/And/But prefixes](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#behavior-driven-style), 

> Prefixes Given, When, Then, And and But are dropped when matching keywords are searched, if no match with the full name is found

Is there a builtin method to reverse this logic such that it checks for the prefix-dropped version first?

```
*** Keywords ***
${url} is open
    Log To Console           Open "${url}"

${ele} is clicked
    Log To Console           Click "${ele}"

${ele} should be visible
    Log To Console           Is Visible "${ele}"


*** Variables ***
${url}          https://google.com
${btn_1}        xpath=//a/b/c
${modal_2}      xpath=//a/b/c


*** Test Cases ***
Valid Login
    Given ${url} is open
    When ${btn_1} is clicked
    Then ${modal_2} should be visible
```

Outputs
```
Open "Given https://google.com"
Click "When xpath=//a/b/c"
Is Visible "Then xpath=//a/b/c"
```

Wanted output
```
Open "https://google.com"
Click "xpath=//a/b/c"
Is Visible "xpath=//a/b/c"
```

---

I can get this behaviour by changing https://github.com/robotframework/robotframework/blob/master/src/robot/running/namespace.py#L292-L298 to 

```python
        runner = self._get_runner_from_suite_file(name)
        if not runner and '.' in name:
            runner = self._get_explicit_runner(name)
        if not runner:
            runner = self._get_bdd_style_runner(name)
        if not runner:
            runner = self._get_implicit_runner(name)
```

**Comment by pekkaklarck** [comment]
There's PR #5096 implementing this functionality so let's include this issue into RF 7.1 scope. I won't have time to review the PR right now, but I return to it after RF 7.0.1 is out.

**Comment by pekkaklarck** [comment]
There are some backwards incompatibility concerns due to behavior being changed. In my opinion risks are small enough so that this doesn't need to wait until a major version, but the change needs to be nevertheless be mentioned in release notes.

**Comment by pekkaklarck** [comment]
Implemented in PR #5096 by @JFoederer. Thanks for the contribution!
